swayfmt/src/comments.rs

```diff
@@ -0,0 +1,67 @@
+use std::{fmt::Write, ops::Range};
+use sway_ast::token::Comment;
+use sway_types::Spanned;
+
+use crate::{formatter::FormattedCode, Formatter, FormatterError};
+
+/// Given a range, return an iterator to comments contained within the range.
+pub fn comments_between<'a>(
+    range: &'a Range<usize>,
```

Comment:
```suggestion
    range: Range<usize>,
```
`Range` is just copy data and its more common to take it by value. That said, `Range` notably doesn't actually implement `Copy`! The reason `Copy` was omitted for `Range` is that `Range` is also an `Iterator` type, and its easy to make confusing mistakes during iteration when `Iterator`s are `Copy` (e.g. it's easy to accidentally call `.next()` on a copy rather than the original iterator, causing infinite loops or unexpected results).

Reply:
correct me if there's a better way but I'm taking to take a reference here because `range` is being used within the `filter_map()` closure (which is trying to capture its value). Would its usage here in this case be a footgun?